### PR TITLE
API-15548: Person fhir model

### PR DESCRIPTION
**Pull request by jabando-va**

### CODE REVIEW

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Person.java [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+@Builder
+@Jacksonized
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+@AllArgsConstructor
+@JsonAutoDetect(
+    fieldVisibility = JsonAutoDetect.Visibility.ANY,
+    isGetterVisibility = JsonAutoDetect.Visibility.NONE)
+@Schema(
+    description = "https://www.hl7.org/fhir/R4/person.html",
+    example = "${r4.person:gov.va.api.health.r4.api.swaggerexamples" + ".SwaggerPerson#person}")
```
Comment (jblefkove-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Person.java:53]: Do you want to add this swagger example?

Comment (jhulbert-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Person.java:53]: What's up with the catting of strings on a single line?

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Person.java [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+  @Valid Gender gender;
+
+  @Pattern(regexp = Fhir.DATE)
+  String birthDate;
+
+  @Valid List<Address> address;
+
+  @Valid CodeableConcept maritalStatus;
+
+  @Valid List<Attachment> photo;
```
Comment (jhulbert-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Person.java:96]: Photo is not a list

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Person.java [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+  @Valid List<ContactPoint> telecom;
+
+  @Valid Gender gender;
+
+  @Pattern(regexp = Fhir.DATE)
+  String birthDate;
+
+  @Valid List<Address> address;
+
+  @Valid CodeableConcept maritalStatus;
```
Comment (jhulbert-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Person.java:94]: I don't see this in the model

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Person.java [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+
+  @Valid List<Attachment> photo;
+
+  @Valid Reference managingOrganization;
+
+  @Valid List<Communication> communication;
```
Comment (jhulbert-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Person.java:100]: I dont see this in model either.

**Review by jhulbert-va — DISMISSED** [review]
LGTM

File: us-core-r4-examples/src/main/java/gov/va/api/health/r4/api/swaggerexamples/SwaggerPerson.java [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+                            .build())
+                    .system("http://va.gov/mpi")
+                    .value("1010101010V666666")
+                    .assigner(Reference.builder().display("Master Patient Index").build())
+                    .build(),
+                Identifier.builder()
+                    .system("http://hl7.org/fhir/sid/us-npi")
+                    .value("9876543210")
+                    .build()))
+        .active(Boolean.TRUE)
```
Comment (jblefkove-va) [inline us-core-r4-examples/src/main/java/gov/va/api/health/r4/api/swaggerexamples/SwaggerPerson.java:43]: use the same order of fields here as in the class-definition/fhir-spec

File: us-core-r4/src/test/java/gov/va/api/health/r4/api/samples/SamplePersons.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        .id("1234")
+        .meta(meta())
+        .implicitRules("https://HelloRules.com")
+        .language("Hello language")
+        .text(narrative())
+        .contained(List.of(resource()))
+        .extension(List.of(extension(), extension()))
+        .modifierExtension(List.of(extension(), extensionWithQuantity(), extensionWithRatio()))
+        .identifier(identifier().asList())
+        .active(true)
```
Comment (jblefkove-va) [inline us-core-r4/src/test/java/gov/va/api/health/r4/api/samples/SamplePersons.java:30]: ditto the field order here

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Person.java [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+
+  @Valid Meta meta;
+
+  @Pattern(regexp = Fhir.URI)
+  String implicitRules;
+
+  @Pattern(regexp = Fhir.CODE)
+  String language;
+
+  // Ancestor -- DomainResource
```
Comment (jblefkove-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Person.java:68]: I think we've established that all these resources implement `Resource` and `DomainResource` by now... this comment can be removed, with line 77 as well

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Person.java [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+
+  @Data
+  @NoArgsConstructor
+  @EqualsAndHashCode(callSuper = true)
+  @JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+  @JsonTypeInfo(use = JsonTypeInfo.Id.NONE, defaultImpl = Person.Bundle.class)
+  @Schema(
+      name = "PersonBundle",
+      example =
+          "${r4.relatedPersonBundle:gov.va.api.health.r4.api.swaggerexamples"
```
Comment (jblefkove-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Person.java:130]: copy-pasta

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Person.java [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+    level2,
+    level3,
+    level4
+  }
+
+  @Data
+  @NoArgsConstructor
+  @EqualsAndHashCode(callSuper = true)
+  @JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+  @JsonTypeInfo(use = JsonTypeInfo.Id.NONE, defaultImpl = Person.Bundle.class)
```
Comment (jblefkove-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Person.java:126]: You don't need to qualify `Bundle` and `Entry` within the class

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Person.java [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+  }
+
+  public enum Gender {
+    male,
+    female,
+    other,
+    unknown
+  }
+
+  public enum Assurance {
```
Comment (jblefkove-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Person.java:115]: alphabetize the order of the classes and enums by their names

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Person.java [hunk 11]
```diff
@@ -0,0 +1,206 @@
...
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import lombok.extern.jackson.Jacksonized;
+
+@Data
+@Builder
+@Jacksonized
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+@AllArgsConstructor
```
Comment (jblefkove-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Person.java:46]: include `(access = AccessLevel.PRIVATE)` here as well

**Review by jblefkove-va — APPROVED** [review]
:shipit:

**Pull request merged**